### DISCUSSION TITLE
How can I wallbash cursor IDE, (vs-code interface)

### DISCUSSION THREAD

**Discussion by KitsuneKode** [prasanthrangan/hyprdots#1831] [Q&A] [answered]
So the issue here is that wallbash works on code perfectly like it is made to be! changes on theme change and on reloads!

But on cursor, although the first theme which was available during installation remains as the theme throughout, regardless of the theme switch and reloads!

I have tried manually installing the vsix, doesn't change anything! 

expected behaviour: change theme on reloads for different themes

@kRHYME7 help me out here

**Comment by kRHYME7** [comment]
> But on cursor, although the first theme which was available during installation remains as the theme throughout, 


What do you mean by Cursor? 
uhm ide? this ? https://www.cursor.com/

**Reply by kRHYME7** [reply]
I see. The fundamental question here is how can we theme cursor? 

Can you search for any method on using pywal for "cursor" or theming "cursor"

**Reply by kRHYME7** [reply]
I see I'll try if cursor can use wallbash for theming

**Reply by kRHYME7** [reply]
@KitsuneKode  

Yeah I think we can support cursor too but for now with manual intervention of installing the theme https://www.cursor.com/how-to-install-extension

We just need to modify the script a little bit. 
![image](https://github.com/user-attachments/assets/b4f370fb-d6ac-43a9-81d1-5e90fd979951)

**Reply by kRHYME7** [reply, marked as answer]
Test it 
change your `  ~/.local/share/bin/wallbashcode.sh `  with this 
```
#!/usr/bin/env sh

#// set variables
scrDir="$(dirname "$(realpath "$0")")"
source "${scrDir}/globalcontrol.sh"
readarray -t codeConf < <(find "${confDir}" -mindepth 1 -maxdepth 1 -type d -name "Code*" -o -name "VSCodium*" -o -name "Cursor*" | sort)
readarray -t codeVsix < <(find "$HOME" -mindepth 1 -maxdepth 1 -type d -name ".vscode*" -o -name ".cursor" | sort)
tmpFile="/tmp/$(id -u)$(basename ${0}).tmp"
tgtFile="extensions/undefined_publisher.wallbash-0.0.1/themes/wallbash-color-theme.json"

#// install  ext

for i in "${!codeVsix[@]}" ;do
    if [ ! -f "${codeVsix[i]}/${tgtFile}" ] ; then
        [ -f "${cacheDir}/landing/Code_Wallbash.vsix" ] || curl -L -o "${cacheDir}/landing/Code_Wallbash.vsix" https://github.com/prasanthrangan/hyprdots/raw/main/Source/arcs/Code_Wallbash.vsix
        case ${codeVsix[i]} in
        *".code"*)
            pkg_installed code && code --install-extension "${cacheDir}/landing/Code_Wallbash.vsix"
            ;;
        *".vscodium"*)
            pkg_installed vscodium && vscodium --install-extension "${cacheDir}/landing/Code_Wallbash.vsix"
            ;;
        *".cursor"*)
            echo "[wallbashcode] Cursor IDE: Manual intervention required for extension installation."
            echo "[wallbashcode] Read the instructions here: https://www.cursor.com/how-to-install-extension "
            ;;
        esac
    fi
    [ -d "$(dirname "${codeVsix[i]}/${tgtFile}")" ] && cp "${cacheDir}/landing/wallbashcode.json" "${codeVsix[i]}/${tgtFile}"
done

#// apply theme

for i in "${!codeConf[@]}" ; do
    [ -f "${codeConf[i]}/User/settings.json" ] || continue
    extTheme="$(jq -r '.["workbench.colorTheme"]' "${codeConf[i]}/User/settings.json")"

    if [ "${extTheme}" != "wallbash" ] ; then
        jq '.["workbench.colorTheme"] = "wallbash"' "${codeConf[i]}/User/settings.json" > "${tmpFile}" && mv "${tmpFile}" "${codeConf[i]}/User/settings.json"
    fi
done

```

run ` wallbashcode.sh ` you'll see this
![image](https://github.com/user-attachments/assets/41dbde37-0973-4903-9063-5c5a839c68a9)

Visit the link and navigate to this directory  `/home/khing/.cache/hyde/landing/Code_Wallbash.vsix  ` this is the vsix fie 

![image](https://github.com/user-attachments/assets/6df68001-350c-4acf-8c9b-c9cecb5e6dff)

**Reply by KitsuneKode** [reply]
@kRHYME7  so sorry for the late response! I actually did installed the vsix file!
I have  exactly done what u have said before but the vsix file was from the hyprdots repo codeosshelp branch!
But it didn't work! 

I now redid everything, after a fresh install, now it seems to be working! Can't really explain what went wrong earlier!

one more thing,
wallbashcode.sh doesnot run due to parsing error!

![image](https://github.com/user-attachments/assets/4d1928ae-19cd-4931-894d-8f00de8b0bde)

Btw your theme looks good, is it a custom?

**Reply by KitsuneKode** [reply]
Thanks  a lot! I @kRHYME7

**Reply by kRHYME7** [reply]
> Btw your theme looks good, is it a custom?

What??? lol just my custom Red theme plus a different wallpaper to show the color difference


>   wallbashcode.sh doesnot run due to parsing error!

try to check/reload cursor IDE  That error is fine.

**Reply by KitsuneKode** [reply]
> What??? lol just my custom Red theme plus a different wallpaper to show the color difference
>>
You mean the Red Stone theme from your gallery right. The border glow is good,! how to increase the radius/size of that glow for any theme?  


> try to check/reload cursor IDE That error is fine.
>>
It works fine, I checked everything!

**Reply by KitsuneKode** [reply]
> @KitsuneKode
> 
> Yeah I think we can support cursor too but for now with manual intervention of installing the theme https://www.cursor.com/how-to-install-extension
> 
> We just need to modify the script a little bit. ![image](https://private-user-images.githubusercontent.com/53417443/366892268-b4f370fb-d6ac-43a9-81d1-5e90fd979951.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNjAyNTcsIm5iZiI6MTcyNjE1OTk1NywicGF0aCI6Ii81MzQxNzQ0My8zNjY4OTIyNjgtYjRmMzcwZmItZDZhYy00M2E5LTgxZDEtNWU5MGZkOTc5OTUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDE2NTIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlMDNmMjhiOTEwODVmNDljNGNiYjcwMjc3YmM2ZjRkMWY3NWZlNTFjYTBkZjc3Yjc3MDdlMzFkNTNjMWM3MTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vORVXPOrsfwtpwsJu9Ptu6gn2wYk9ImYGYdI4DBMBDc)
>>
was talking about this!

**Comment by kRHYME7** [comment]
BTW ` Hyde code theme "Theme-name" ` is supported now in the CLI 

also, can you make a PR with this https://github.com/prasanthrangan/hyprdots/discussions/1831#discussioncomment-10626694 

It's fine to support this I guess

**Reply by KitsuneKode** [reply]
> BTW `Hyde code theme "Theme-name"` is supported now in the CLI
> >
Oh, that's nice, I will check it out! 

I am not sure if there is a notification on new updates when updates come out on Hyde! 
If not like after Hyde upgrade, list out the new features after the update.


> also, can you make a PR with this [#1831 (reply in thread)](https://github.com/prasanthrangan/hyprdots/discussions/1831#discussioncomment-10626694)
> 
> It's fine to support this I guess

Yea sure, just to be sure, PR on the hyprdots repo right? not the Hyde one as its the wrapper around this repo?

**Reply by kRHYME7** [reply]
Yes on this repo. You can modify the wallbashcode.sh. 


About the notification I always try to announce it on Discord. Other than that Github commits 

so updating the CLI is just `Hyde update ` or using the aur `yay - Sy hyde-cli-git` 
But updating the Dots users should run `Hyde restore Config ` for the dotfiles only. Or `Hyde upgrade` if users want to update the new dependencies.

**Reply by KitsuneKode** [reply]
oh I see, so using Hyde restore Config, is more like the optimal way to be updated! Uhm, i used to do Hyde upgrade every time, Hyde upgrade seems to have a slight possibility of breaking the running system! I mean its arch after all,

**Reply by kRHYME7** [reply]
Yup it's the `restore Config ` for restoring just the dots.

The "restore" convention is derived from Users restoring their configuration from github source. It's sometimes confusing from the user's perspective though.

**Reply by KitsuneKode** [reply]
Yea rt, if u hadn't specifically mentioned this to me rn, I would have not known about that, I thought it was just to roll back to previous configs! so never actually dug deeper to it

**Reply by KitsuneKode** [reply]
@kRHYME7 
Done with the PR

Check it!
https://github.com/prasanthrangan/hyprdots/pull/1837
